### PR TITLE
Made changes to support setting mtime and getting when buffered writes are enabled

### PR DESCRIPTION
**Pull request by vadlakondaswetha**
### Description
Updated SetInodeattributes and getAttributes for handling mtime for local files and empty GCS files.
invoking bufferedWrites flow for empty gcs files.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - Added
3. Integration tests - NA

### CODE REVIEW

File: internal/fs/inode/file.go [hunk 1]
```diff
@@ -414,6 +414,14 @@ func (f *FileInode) Attributes(
 		}
 	}
 
+	bwEnabled, err := f.shouldTriggerBufferedWrites()
+	if err != nil {
+		return
+	}
+	if bwEnabled {
+		attrs.Mtime = f.bwh.WriteFileInfo().Mtime
```
Comment (ashmeenkaur) [inline internal/fs/inode/file.go:422]: Should we take size also from bwh?

File: internal/fs/inode/file.go [hunk 2]
```diff
@@ -681,17 +706,28 @@ func (f *FileInode) CreateEmptyTempFile() (err error) {
...
 		}
+		f.bwh.SetMtime(f.mtimeClock.Now())
 	}
 
-	err = f.bwh.Write(data, offset)
+	return nil
+}
 
-	return
+func (f *FileInode) shouldTriggerBufferedWrites() (bool, error) {
```
Comment (ashmeenkaur) [inline internal/fs/inode/file.go:722]: Nit
```suggestion
func (f *FileInode) bufferedWritesEnabled() (bool, error) {
```

File: internal/fs/fs.go [hunk 3]
```diff
@@ -1685,13 +1685,10 @@ func (fs *fileSystem) createLocalFile(
...
-	// For buffered writes, we don't create temp files on disk.
-	if !fs.newConfig.Write.ExperimentalEnableStreamingWrites {
-		// Empty file is created to be able to set attributes on the file.
-		fileInode := child.(*inode.FileInode)
-		if err := fileInode.CreateEmptyTempFile(); err != nil {
-			return nil, err
-		}
+	// Empty file is created to be able to set attributes on the file.
+	fileInode := child.(*inode.FileInode)
+	if err := fileInode.CreateEmptyTempFile(); err != nil {
```
Comment (ashmeenkaur) [inline internal/fs/fs.go:1690]: This looks like a tech debt.. is there no way we can set attributes in bwh?

I think this also allows truncating the file in SetInodeAttributes. We don't want to allow that, right? Shall we throw an error / log a warning explicitly?

File: internal/fs/inode/file.go (hunk 1 above)
Comment (vadlakondaswetha) [reply internal/fs/inode/file.go:422]: Thanks. Since i focussed only on mtime, missed the size. updated now

File: internal/fs/inode/file.go [hunk 4]
```diff
@@ -673,6 +696,15 @@ func (f *FileInode) CacheEnsureContent(ctx context.Context) (err error) {
 }
 
 func (f *FileInode) CreateEmptyTempFile() (err error) {
```
Comment (ashmeenkaur) [inline internal/fs/inode/file.go:698]: Shall we rename this function to `CreateWriteHandler()` or `CreateBufferedOrTempWriter()`

File: internal/fs/inode/file_test.go [hunk 5]
```diff
@@ -1004,6 +1066,59 @@ func (t *FileTest) MultipleWritesToLocalFileWhenStreamingWritesAreEnabled() {
 	err = t.in.Write(t.ctx, []byte("hello"), 0)
 	AssertEq(nil, err)
 	AssertEq(7, t.in.bwh.WriteFileInfo().TotalSize)
+	// The inode should agree about the new mtime.
```
Comment (ashmeenkaur) [inline internal/fs/inode/file_test.go:1069]: shall we check for mtime too? same for WriteToEmptyGCSFileWhenStreamingWritesAreEnabled

File: internal/fs/inode/file.go [hunk 6]
```diff
@@ -673,6 +696,15 @@ func (f *FileInode) CacheEnsureContent(ctx context.Context) (err error) {
...
+	// Skip creating empty file when streaming writes are enabled
+	if f.local && f.writeConfig.ExperimentalEnableStreamingWrites {
```
Comment (ashmeenkaur) [inline internal/fs/inode/file.go:700]: Should this be for size 0 too?

Comment (vadlakondaswetha) [reply internal/fs/inode/file.go:700]: This method is invoked only in CreateFile flow and not open file.
Size 0 files come in open file flow.

**Review by ashmeenkaur — APPROVED** [review]

**Pull request merged**